__tests__/AmplifyMapLibreRequest.test.ts

```diff
@@ -63,6 +63,26 @@ describe("AmplifyMapLibreRequest", () => {
     expect(request.url).toContain("x-amz-user-agent");
   });
 
+  test("transformRequest appends query params to existing query params if any", () => {
```

Comment:
can we add an explicit check to this test that we don't have multiple `?` 

Can we also add test cases for edge cases like
- nothing after the question mark
- nothing after an ampersand
- query params with ampersands

Otherwise change looks good though :)

Reply:
Added tests to cover these cases:
- [x] nothing after the question mark
- [x] nothing after an ampersand
- [x] query params with ampersands